mitiq/pec/decomposition/optimal.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2020 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from typing import List, Tuple
+
+from cirq import I, Operation, X, Y, Z
+
+PAULIS = [I, X, Y, Z]
+
+
+def depolarizing_decomposition(
+    ideal_operation: Operation, noise_level: float
+) -> List[Tuple[float, List[Operation]]]:
+    r"""As described in [Temme2017]_, optimally decompose a single-qubit
+    ``ideal_operation`` :math:`\mathcal{U}_{\beta}` into its quasi-probability
+    representation (QPR), which is a linear combination of noisy implementable
+    operations :math:`\{\eta_{\alpha} \mathcal{O}_{\alpha}\}`. Takes as an
+    assumption that the basis of implementable operations includes any desired
+    ideal operation followed by a depolarizing channel (meaning that all
+    :math:`\mathcal{O}_{\alpha} = \mathcal{D} \circ \mathcal{U}`, where
+    :math:`\mathcal{D}(\rho) =  (1 - \epsilon) \rho + \epsilon I/2`). Given
```

Comment:
The `/2` factor after the identity is valid only for a single qubit but the function is more general. For n qubits the normalization is 1/(2^n).